parser/src/sourcemanager.cpp

```diff
@@ -221,6 +234,31 @@ void SourceManager::updateFile(const model::File& file_)
     });
 }
 
+void SourceManager::removeFile(const model::File& file_)
+{
+  bool removeContent = false;
+
+  // Delete File and FileContent (only when no other File references it)
+  _transaction([&, this]() {
+    auto relFiles = _db->query<model::File>(
+        odb::query<model::File>::content == file_.content.object_id());
+    if(std::distance(relFiles.begin(), relFiles.end()) == 1)
+    {
+      removeContent = true;
+      _db->erase<model::FileContent>(file_.content.object_id());
+    }
+    _db->erase<model::File>(file_.id);
+  });
+
+  // Maintain cache
+  _createFileMutex.lock();
+  _files.erase(file_.path);
+  _persistedFiles.erase(file_.id);
+  if(removeContent)
```

Comment:
Space missing.

Reply:
Solved in 4b52c300da8e3f8309356cf71e5f9c70207cc051